### PR TITLE
Add Travis CI for automated linting and testing

### PR DESCRIPTION
**Pull request by nicholaschuayunzhi**
Related https://github.com/MarkBind/markbind/issues/121

This pr introduces a script that builds a site based on the recently merged test site #27.

# How it works
1. run ESLint
1. build site with local `markbind-cli` and `npm published markbind`
1.  compare _site with expected site
    - the script ignores diffs which are different path separators for html files
1. if there were any errors, the build fails

The ignore path separator logic was written by me. I made it very specific so its likely there will be more false positives than negatives.

NOTE: need to set up Travis to run on this repo.

Tested on my fork:
https://github.com/nicholaschuayunzhi/markbind-cli/pull/3

### CODE REVIEW

File: test/test_site/testUtil/diffHtml.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
+const jsdiff = require('diff');
+
+/**
+ * Checks if fragment ends with a an unclosed path
```
Comment (acjh) [inline test/test_site/testUtil/diffHtml.js]: ~`a`~

**Review by acjh — COMMENTED** [review]
Looks good.

**Comment by acjh** [comment]
> 

Are we able to *reliably* get the line number where the diff occurs?

File: package.json [hunk 2]
```diff
@@ -23,6 +23,7 @@
     "chokidar": "^1.6.1",
     "clear": "^0.0.1",
     "commander": "^2.9.0",
+    "diff": "^3.5.0",
```
Comment (Gisonrg) [inline package.json]: If this is only used in testing, put it under `devDependencies`

File: test/test_site/test.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#! /bin/bash
```
Comment (Gisonrg) [inline test/test_site/test.sh]: no need space here

Comment (Gisonrg) [inline test/test_site/test.sh]: We should add comments for bash script :) It is not easy to read

File: test/test_site/testUtil/test.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    throw new Error('Different files built');
+  }
+
+  // check contents
+  const resolvedExpectedFilePath = path.resolve('./expected', expectedFilePath);
+  const resolvedActualFilePath = path.resolve('./_site', actualFilePath);
+  const expected = fs.readFileSync(resolvedExpectedFilePath, 'utf8');
+  const actual = fs.readFileSync(resolvedActualFilePath, 'utf8');
+  if (path.parse(resolvedActualFilePath).ext !== '.html') {
+    if (expected !== actual) {
```
Comment (Gisonrg) [inline test/test_site/testUtil/test.js]: I am not sure if we want to compare asset files...what if you have two binary files?

Maybe just compare html files for now?

**Comment by nicholaschuayunzhi** [comment]
> Are we able to reliably get the line number where the diff occurs?

I'm currently looking into it. There doesn't seem to be an easy way to do it as of yet.

**Comment by acjh** [comment]
@Gisonrg Shall we merge this first?

You need to "Activate" Travis via https://travis-ci.org/MarkBind/markbind-cli as the repository owner.

**Comment by Gisonrg** [comment]
Yea we can merge this first to see how it works :) I'll set up the Travis tomorrow.

**Review by Gisonrg — APPROVED** [review]

**Comment by rachx** [comment]
Remember to update the developer guide (e.g. testing locally)

**Pull request merged**